ioengine/ioengine.py

```diff
@@ -0,0 +1,63 @@
+"""The IOEngine class."""
+import inspect
+import sys
+from types import ModuleType
+
+from ioengine.utils import dotdict
+from ioengine.exceptions import UnsupportedAPI
+
+
+class IOEngine:
+    """The IOEngine class."""
+
+    def __init__(self):
+        """Set up engine."""
+        self.services = []
+        self.engine_api = dotdict(showMessage=print, register=self.register_service)
+
+        # make a fake module with api
+        mod = ModuleType("bioimage")
+        sys.modules[mod.__name__] = mod  # pylint: disable=no-member
+        mod.__file__ = mod.__name__ + ".py"  # pylint: disable=no-member
+        mod.api = self.engine_api
+
+    def register_service(self, **api):
+        """Set interface."""
+        if isinstance(api, dict):
+            api = {a: value for a, value in api.items() if not a.startswith("_")}
+        elif inspect.isclass(type(api)):
+            api = {a: getattr(api, a) for a in dir(api) if not a.startswith("_")}
+        else:
+            raise UnsupportedAPI
```

Comment:
I'm trying to write a test for this, but I don't think we will hit this case like this. We'll error before that if we pass something else than a dict to `register_service`.

Reply:
So here I think it's important to support both a dictionary and a class/object.

Reply:
Yes. How can we support passing a class instance? We're just accepting keyword arguments now. That works with a `dict`. Will it work with an arbitrary instance?